src/dotnet/ToolManifest/ToolManifestEditor.cs

```diff
@@ -217,6 +218,44 @@ private class SerializableLocalToolsManifest
             [JsonProperty(Required = Required.Always)]
             // The dictionary's key is the package id
             public Dictionary<string, SerializableLocalToolSinglePackage> tools { get; set; }
+
+            public SerializableLocalToolsManifest With(Dictionary<string, SerializableLocalToolSinglePackage> newTools)
```

Comment:
Perhaps this should be called `WithTools`?  `With` alone isn't very descriptive of what is being added.

Reply:
Renamed to `NewCopyWithTools`